wcfsetup/install/files/lib/system/html/input/node/HtmlInputNodeImg.class.php

```diff
@@ -365,4 +366,29 @@ private function getParentFigure(\DOMElement $img): ?\DOMElement
 
         return null;
     }
+
+    /**
+     * Ckeditor will wrap the `<a>` tag into a `<figure>` tag.
+     * This method moves the `<a>` tag out of the `<figure>` tag, to keep the link available.
+     *
+     * So the new structure is:
+     *  <a>
+     *      <figure>
+     *          <img>
+     *      </figure>
+     * </a>
+     */
+    private function moveFigureIntoATag(\DomElement $img): void
+    {
+        $figure = $this->getParentFigure($img);
+        $parent = $img->parentElement;
+        \assert($parent instanceof \DOMElement);
+        if ($figure === null || $parent->nodeName !== 'a') {
+            return;
+        }
+        $figure->parentNode->insertBefore($parent, $figure);
+        $parent->removeChild($img);
```

Comment:
```suggestion
```

This is redundant and possibly misleading. DOM nodes can only have one parent at a time, appending it somewhere else will move the entire node, regardless of where it was before.